src/cmd/shard/main.go

```diff
@@ -20,31 +21,45 @@ func main() {
 }
 
 func do() error {
-	if len(os.Args) != 3 {
-		return fmt.Errorf("unknown args: %v", os.Args)
-	}
-	shardStr := os.Args[1]
-	address := os.Args[2]
-
-	shardNum, modulos, err := route.ParseShard(shardStr)
+	shardNum := flag.Int("shard", -1, "Optional. The shard to service.")
+	modulos := flag.Int("modulos", 4, "The total number of shards.")
+	address := flag.String("address", "", "Optional. The address to advertise for this node.")
+	flag.Parse()
+	addrs, err := net.InterfaceAddrs()
 	if err != nil {
 		return err
 	}
+	if *address == "" {
+		// No address, we'll try to use our ip addr instead
+		for _, addr := range addrs {
+			if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+				if ipnet.IP.To4() != nil {
+					*address = ipnet.IP.String()
+					break
+				}
+			}
+		}
+	}
+	if *address == "" {
+		return fmt.Errorf("pfs: Couldn't find machine ip.")
+	}
+
 	shard := storage.NewShard(
-		"http://"+address,
-		"data-"+shardStr,
-		"comp-"+shardStr,
-		"pipe-"+shardStr,
-		shardNum,
-		modulos,
+		"http://"+*address,
+		fmt.Sprintf("data-%d-%d", *shardNum, *modulos),
+		fmt.Sprintf("pipe-%d-%d", *shardNum, *modulos),
+		uint64(*shardNum),
```

Comment:
ah whatever, it bugs me, dont want guaranteed overflow situations committed, but i think we'll clean this up anyways and if it works, all good